src/test/e2e/29_mismatched_checks_test.go

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package test provides e2e tests for Zarf.
+package test
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/stretchr/testify/require"
+)
+
+// TestMismatchedArchitectures ensures that zarf produces an error
+// when the package architecture doesn't match the target cluster architecture.
+func TestMismatchedArchitectures(t *testing.T) {
+	t.Log("E2E: Mismatched architectures")
+	e2e.SetupWithCluster(t)
+
+	var (
+		mismatchedArch         = e2e.GetMismatchedArch()
+		mismatchedGamesPackage = fmt.Sprintf("zarf-package-dos-games-%s-1.0.0.tar.zst", mismatchedArch)
+		expectedErrorMessage   = fmt.Sprintf("this package architecture is %s", mismatchedArch)
+	)
+
+	// Build dos-games package with different arch than the cluster arch.
+	stdOut, stdErr, err := e2e.Zarf("package", "create", "examples/dos-games/", "--architecture", mismatchedArch, "--confirm")
+	require.NoError(t, err, stdOut, stdErr)
+	defer e2e.CleanFiles(mismatchedGamesPackage)
+
+	// Ensure zarf package deploy returns an error because of the mismatched architectures.
+	_, stdErr, err = e2e.Zarf("package", "deploy", mismatchedGamesPackage, "--confirm")
+	require.Error(t, err, stdErr)
+	require.Contains(t, stdErr, expectedErrorMessage)
+}
+
+// TestMismatchedVersions ensures that zarf produces a warning
+// when the initialized version of Zarf doesn't match the current CLI
+func TestMismatchedVersions(t *testing.T) {
+	t.Log("E2E: Mismatched versions")
+	e2e.SetupWithCluster(t)
+
+	var (
+		expectedWarningMessage = "Potential Breaking Changes"
+	)
+
+	// Get the current init package secret
+	initPkg := types.DeployedPackage{}
+	base64Pkg, _, err := e2e.Kubectl("get", "secret", "zarf-package-init", "-n", "zarf", "-o", "jsonpath={.data.data}")
+	require.NoError(t, err)
+	jsonPkg, err := base64.StdEncoding.DecodeString(base64Pkg)
+	require.NoError(t, err)
+	fmt.Println(string(jsonPkg))
+	err = json.Unmarshal(jsonPkg, &initPkg)
+	require.NoError(t, err)
```

Comment:
we could replace this w/ `zarf package inspect init -o json` if we wanted to

Reply:
we actually can't because this is a DeployedPackage not a ZarfPackage.

Reply:
Then `zarf package list -o json --package=init` could work, no?